src/subscriptions.c

```diff
@@ -952,6 +954,24 @@ subscription_done(void)
  * *************************************************************************/
 
 /**
+ * Update incoming byte count
+ */
+void subscription_add_bytes_in(th_subscription_t *s, size_t in)
+{
+  atomic_add(&s->ths_bytes_in, in);
+  atomic_add_u64(&s->ths_total_bytes_in, in);
```

Comment:
OK. I think that my explanation was not understandable. I would keep only atomic_add_u64(&s->ths_total_bytes_in, in); in subscription_add_bytes_in() and change subscription_status_callback() as proposed - new exchange code.